homeassistant/components/nest/__init__.py

```diff
@@ -166,38 +166,43 @@ async def async_handle_event(self, event_message: EventMessage) -> None:
         )
         if not device_entry:
             return
+        supported_traits = self._supported_traits(device_id)
         for api_event_type, image_event in events.items():
             if not (event_type := EVENT_NAME_MAP.get(api_event_type)):
                 continue
             nest_event_id = image_event.event_token
-            attachment = {
-                "image": EVENT_THUMBNAIL_URL_FORMAT.format(
-                    device_id=device_entry.id, event_token=image_event.event_token
-                ),
-            }
-            if self._supports_clip(device_id):
-                attachment["video"] = EVENT_MEDIA_API_URL_FORMAT.format(
-                    device_id=device_entry.id, event_token=image_event.event_token
-                )
             message = {
                 "device_id": device_entry.id,
                 "type": event_type,
                 "timestamp": event_message.timestamp,
                 "nest_event_id": nest_event_id,
-                "attachment": attachment,
             }
+            if (
+                TraitType.CAMERA_EVENT_IMAGE in supported_traits
+                or TraitType.CAMERA_CLIP_PREVIEW in supported_traits
+            ):
+                attachment = {
+                    "image": EVENT_THUMBNAIL_URL_FORMAT.format(
+                        device_id=device_entry.id, event_token=image_event.event_token
+                    )
+                }
+                if TraitType.CAMERA_CLIP_PREVIEW in supported_traits:
+                    attachment["video"] = EVENT_MEDIA_API_URL_FORMAT.format(
+                        device_id=device_entry.id, event_token=image_event.event_token
+                    )
+                message["attachment"] = attachment
             if image_event.zones:
                 message["zones"] = image_event.zones
             self._hass.bus.async_fire(NEST_EVENT, message)
 
-    def _supports_clip(self, device_id: str) -> bool:
+    def _supported_traits(self, device_id: str) -> list[TraitType]:
         if not (
             device_manager := self._hass.data[DOMAIN]
             .get(self._config_entry_id, {})
             .get(DATA_DEVICE_MANAGER)
         ) or not (device := device_manager.devices.get(device_id)):
-            return False
-        return TraitType.CAMERA_CLIP_PREVIEW in device.traits
+            return []
```

Comment:
Missing coverage on this line

Reply:
This is complex to test because it involves a race, and is a bit theoretical. This can happen when there is a race on initialization where a pubsub message is received before the device manager has been registered.

The subscriber is started first and registers this callback. Then the device manager is initialized and registered.  it happens in this order because otherwise there is a race where the device manager will fetch initial state, then a message is published which isn't received, and we are missing state. Therefore we first subscribe and then fetch initial state.

Adding coverage for this could involve patching and making the device manager block, then sending messages and verifying they are received -- but the integration wouldn't have finished setup by then so its kind of odd.

If you have suggestions for a different approach i am open to it.